include/s3select_csv_parser.h

```diff
@@ -1,413 +1,418 @@
-#include <iostream>
-
-#include <boost/mpl/vector/vector30.hpp>
-// back-end
-#include <boost/msm/back/state_machine.hpp>
-//front-end
-#include <boost/msm/front/state_machine_def.hpp>
-
-#include <vector>
-
-namespace msm = boost::msm;
-namespace mpl = boost::mpl;
-
-namespace s3selectEngine
-{
-// events
-struct event_column_sep {};
-struct event_eol {};
-struct event_end_of_stream {};
-struct event_not_column_sep {};//i.e any char
-struct event_quote {};
-struct event_escape {};
-struct event_empty {};
-
-
-// front-end: define the FSM structure
-struct csvStateMch_ : public msm::front::state_machine_def<csvStateMch_>
-{
-  char* input_stream;
-  std::vector<char*>* tokens;
-  std::vector<int> has_esc{128};
-  size_t token_idx;
-  size_t escape_idx;
-  char* input_cur_location;
-  char* start_token;
-  bool end_of_parse;
-
-  typedef csvStateMch_ csv_rules;
-
-  csvStateMch_():end_of_parse(false) {}
-
-  void set(const char* input, std::vector<char*>* tk)
-  {
-    input_cur_location = input_stream = const_cast<char*>(input);
-    token_idx = 0;
-    tokens = tk;
-    escape_idx = 0;
-  }
-
-  char get_char()
-  {
-    return *input_cur_location;
-  }
-
-
-  char get_next_char(const char* end_stream)
-  {
-    input_cur_location++;
-
-    if(input_cur_location >= end_stream)
-    {
-      return 0;
+#include "csvparser/csv.h"
+
+namespace io{
+
+    namespace error{
+        struct escaped_char_missing :
+                base,
+                with_file_name,
+                with_file_line{
+                void format_error_message()const override{
+                        std::snprintf(error_message_buffer, sizeof(error_message_buffer),
+                                "Escaped character missing in line %d in file \"%s\"."
+                                , file_line, file_name);
+                }
+        };
     }
 
-    return *input_cur_location;
-  }
-
-  const char* currentLoc()
-  {
-    return input_cur_location;
-  }
-
-  void parse_escape(char* in, char esc_char='\\')
-  {
-    //assumption atleast one escape and single
-    char* dst, *src;
-
-    dst = src = in;
-
-    while (1)
-    {
-      while (*src && *src != esc_char)
-      {
-        src++;  //search for escape
-      }
-
-      if (!*src) //reach end
-      {
-        char* p = src;
-        while (dst < src)
+    namespace detail{
+        static void unescape(char*&col_begin, char*&col_end, char& quote, char& escape_char)
```

Comment:
in case escape-char and quote-char are not defined (empty)
to skip this functionality